api/users/serializers.py

```diff
@@ -128,6 +132,40 @@ def profile_image_url(self, user):
         size = self.context['request'].query_params.get('profile_image_size')
         return user.profile_image_url(size=size)
 
+    def _validate_user_json(self, value, json_schema):
+        try:
+            jsonschema.validate(value, from_json(json_schema))
+        except jsonschema.ValidationError as e:
+            if len(e.path) > 1:
+                raise InvalidModelValueError("For '{}' the field value {}".format(e.path[-1], e.message))
+
+            raise InvalidModelValueError(e.message)
+        except jsonschema.SchemaError as e:
+            raise InvalidModelValueError(e.message)
+
+        self._validate_dates(value)
+
+    def _validate_dates(self, info):
+        for history in info:
+
+            startDate = datetime.date(history['startYear'], history['startMonth'], 1)
```

Comment:
Discussed above: `startMonth` and `endMonth` aren't required so these lines will throw `KeyError`s